src/Controller/Event/ControllerEvent.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Cake\Controller\Event;
+
+use Cake\Controller\Controller;
+use Cake\Event\Event;
+
+class ControllerEvent extends Event
+{
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param string $name Name of the event
+     * @param Controller $controller the controller that this event applies to
+     * @param array|null $data any value you wish to be transported with this event to it can be read by listeners
+     */
+    public function __construct($name, Controller $controller, $data = null)
+    {
+        if (!$controller instanceof Controller) {
+            throw new \InvalidArgumentException('Controller event should provide the controller name');
```

Comment:
does this exception could actually be raised ? there's a Controller type hinting...